virttest/utils_net.py

```diff
@@ -2387,10 +2438,11 @@ class VirtIface(propcan.PropCan, object):
 
     __slots__ = ['nic_name', 'g_nic_name', 'mac', 'nic_model', 'ip',
                  'nettype', 'netdst']
-    # Make sure first byte generated is always zero and it follows
-    # the class definition.  This helps provide more predictable
-    # addressing while avoiding clashes between multiple NICs.
-    LASTBYTE = random.SystemRandom().randint(0x00, 0xff)
+    # Using MA-S assignment here, that means we can have at most 4096 unique
+    # identifiers (MAC addresses) on the same job instance. We may consider
+    # using bigger blocks for large-scale deployment, such as microVM
+    # applications
+    EUI48_ASSIGNMENT = ieee_eui48_assignment(36, repeat=True)
```

Comment:
shouldn't it be 24 ?

Reply:
Well, it can be. IEEE provides 3 standard EUI assignments: `MA-L`, `MA-M` and `MA-S`. If we choose `MA-L` assignment here the value should be 24, but higher probability of MAC collision will be the side effect  (see https://github.com/avocado-framework/avocado-vt/issues/2121).

Considered that we didn't have the demand of so large scale deployment of VMs, I chose `MA-S` assignment to minimize the probability.

FYI: https://standards.ieee.org/content/dam/ieee-standards/standards/web/documents/tutorials/eui.pdf (EUI Structure and Representation)

Reply:
makes sense.